protocol/x/clob/keeper/process_operations.go

```diff
@@ -248,6 +219,229 @@ func (k Keeper) PersistMatchToState(
+	// case types.OrderRemoval_REMOVAL_REASON_INVALID_REDUCE_ONLY:
+	// 	if !orderToRemove.IsReduceOnly() {
+	// 		return errorsmod.Wrapf(
+	// 			types.ErrInvalidOrderRemoval,
+	// 			"Order Removal (%+v) invalid. Order must be reduce only.",
+	// 			*orderRemoval,
+	// 		)
+	// 	}
+
+	// 	// The reduce-only order must increase or change the side of the position to trigger removal.
+	// 	currentPositionSize := k.GetStatePosition(
+	// 		ctx,
+	// 		orderIdToRemove.SubaccountId,
+	// 		orderToRemove.GetClobPairId(),
+	// 	)
+	// 	orderQuantumsToFill := orderToRemove.GetBigQuantums()
+
+	// 	orderFillWouldIncreasePositionSize := orderQuantumsToFill.Sign() == currentPositionSize.Sign()
+
+	// 	newPositionSize := new(big.Int).Add(currentPositionSize, orderQuantumsToFill)
+	// 	orderChangedSide := currentPositionSize.Sign()*newPositionSize.Sign() == -1
+	// 	if !orderFillWouldIncreasePositionSize && !orderChangedSide {
+	// 		return errorsmod.Wrapf(
+	// 			types.ErrInvalidOrderRemoval,
+	// 			"Order Removal (%+v) invalid. Order fill must increase position size or change side.",
+	// 			*orderRemoval,
+	// 		)
+	// 	}
+	case types.OrderRemoval_REMOVAL_REASON_POST_ONLY_WOULD_CROSS_MAKER_ORDER:
+		// TODO (CLOB-877)
+		// This removal reason implies the stateful taker order could not be placed on the book because it crossed
+		// some unspecified maker order. We must verify that the stateful taker order should have been placed in this block
+		// and that the maker order it crossed with actually exists on the proposer's orderbook. This information is not
+		// currently available.
+		k.statUnverifiedOrderRemoval(ctx, orderRemoval) // remove when above is complete
+
+		// The order should be post-only
+		if orderToRemove.TimeInForce != types.Order_TIME_IN_FORCE_POST_ONLY {
+			return errorsmod.Wrap(
+				types.ErrUnexpectedTimeInForce,
+				"Order is not post-only.",
+			)
+		}
+	case types.OrderRemoval_REMOVAL_REASON_INVALID_SELF_TRADE:
+		// TODO (CLOB-877)
+		// This removal reason implies the maker order was removed during a self-trade encountered during matching.
+		// We must verify that the stateful order had already been placed, and that the taker order matching which caused
+		// this removal would have matched against this order. This information is not currently available.
+		k.statUnverifiedOrderRemoval(ctx, orderRemoval) // remove when above is complete
+	case types.OrderRemoval_REMOVAL_REASON_CONDITIONAL_FOK_COULD_NOT_BE_FULLY_FILLED:
+		// TODO (CLOB-877)
+		// The order must have been triggered in the last block. This information is not currently available.
+		k.statUnverifiedOrderRemoval(ctx, orderRemoval) // removed when above is complete
+
+		// The order should be in triggered state.
+		_, found := k.GetTriggeredConditionalOrderPlacement(ctx, orderIdToRemove)
+		if !found {
+			return errorsmod.Wrap(
+				types.ErrConditionalOrderUntriggered,
+				"Order is not in triggered state.",
+			)
+		}
+
+		// The order should be FOK
```

Comment:
This check can be before the `GetTriggeredConditionalOrderPlacement` check, the first one would not return anything if it wasn't FOK

Reply:
I honestly don't even need the `GetTriggeredConditionalOrderPlacement` check. We already verify the order exists in state using FetchOrderFromOrderId and it requires conditional orders to be in triggered state, so this is redundant.